opbeat/processors.py

```diff
@@ -14,112 +14,210 @@
+    Sanitizes http request headers
+
+    :param client: an Opbeat client
+    :param event: a transaction or error event
+    :return: The modified event
     """
-    def process(self, data, **kwargs):
-        if 'stacktrace' in data:
-            for frame in data['stacktrace'].get('frames', []):
-                frame.pop('vars', None)
+    try:
+        headers = event['context']['request']['headers']
+        event['context']['request']['headers'] = varmap(_sanitize, headers)
+    except (KeyError, TypeError):
+        pass
+    return event
 
-        return data
 
+def sanitize_http_wsgi_env(client, event):
+    """
+    Sanitizes WSGI environment variables
 
-class SanitizePasswordsProcessor(Processor):
+    :param client: an Opbeat client
+    :param event: a transaction or error event
+    :return: The modified event
     """
-    Asterisk out passwords from password fields in frames, http,
-    and basic extra data.
+    try:
+        env = event['context']['request']['env']
+        event['context']['request']['env'] = varmap(_sanitize, env)
+    except (KeyError, TypeError):
+        pass
+    return event
+
+
+def sanitize_http_request_querystring(client, event):
     """
-    MASK = '*' * 8
-    FIELDS = frozenset([
-        'password',
-        'secret',
-        'passwd',
-        'token',
-        'api_key',
-        'access_token',
-        'sessionid',
-    ])
-    VALUES_RE = re.compile(r'^\d{16}$')
+    Sanitizes http request query string
 
-    def sanitize(self, key, value):
-        if value is None:
-            return
+    :param client: an Opbeat client
+    :param event: a transaction or error event
+    :return: The modified event
+    """
+    try:
+        query_string = force_text(
+            event['context']['request']['url']['search'],
+            errors='replace'
+        )
+    except (KeyError, TypeError):
+        return event
+    if '=' in query_string:
+        sanitized_query_string = _sanitize_string(query_string, '&', '=')
+        raw = event['context']['request']['url']['raw']
+        event['context']['request']['url']['search'] = sanitized_query_string
+        event['context']['request']['url']['raw'] = raw.replace(
+            query_string,
+            sanitized_query_string
+        )
+    return event
+
+
+def sanitize_http_request_body(client, event):
+    """
+    Sanitizes http request body. This only works if the request body
+    is a query-encoded string. Other types (e.g. JSON) are not handled by
+    this sanitizer.
 
-        if isinstance(value, six.string_types) and self.VALUES_RE.match(value):
-            return self.MASK
+    :param client: an Opbeat client
+    :param event: a transaction or error event
+    :return: The modified event
+    """
+    try:
+        body = force_text(
+            event['context']['request']['body'],
+            errors='replace'
+        )
+    except (KeyError, TypeError):
+        return event
+    if '=' in body:
```

Comment:
it's a heuristic to try and tell if it's a querystring-encoded string. Not great, but there's probably not much we can do better. Open for suggestions :)